src/handle_commit.cxx

```diff
@@ -269,12 +269,10 @@ void raft_server::commit_app_log(ulong idx_to_commit,
     if (need_to_handle_commit_elem) {
         std::unique_lock<std::mutex> cre_lock(commit_ret_elems_lock_);
         bool match_found = false;
-        auto entry = commit_ret_elems_.begin();
-        while (entry != commit_ret_elems_.end()) {
+        auto entry = commit_ret_elems_.find(sm_idx);
+        if (entry != commit_ret_elems_.end()) {
             ptr<commit_ret_elem> elem = entry->second;
```

Comment:
Actually the changes regarding `commit_ret_elems_lock_` have been made a few months ago so I have already forgotten some details here.  But my main purpose is to avoid locking unnecessary scope to reduce lock contention.

And for `commit_app_log()`, I want to avoid `elem->awaiter_.invoke()` inside `commit_ret_elems_lock_` as it would wait other thread to gain lock inside.

Besides the lock, for `commit_app_log()` function, I have removed the while loop for linear search and instead I use the `commit_ret_elems_.find()` for getting the required entry.  Please review if it's ok.